File: 728-self-dividing-numbers/self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ans = new ArrayList();
        for(int i=left;i<=right;i++){
            int n = i;
            boolean flag = true;

            while(n!=0){
                int a = n%10;
                if(a==0 || i%a!=0){
                    flag = false;
                    break;
                }
                n = n / 10;
            }
            if(flag) ans.add(i);
        }
        return ans;
    }
}

File: 728-self-dividing-numbers/self-dividing-numbers.ts
function selfDividingNumbers(left: number, right: number): number[] {
    let ans = [];
    for(let i=left;i<=right;i++){
        let n = i;
        let flag = true;

        while(n!=0){
            let a = n%10;
            if(i%a!=0){
                flag = false;
                break;
            }
            n = Math.floor(n / 10);
        }

        if(flag) ans.push(i);

    }

    return ans;
    
    
};